src/CodeReport/AnnotateDisassemblyTest.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <absl/strings/str_replace.h>
+#include <gtest/gtest.h>
+
+#include <filesystem>
+
+#include "CodeReport/AnnotateDisassembly.h"
+#include "CodeReport/AnnotatingLine.h"
+#include "CodeReport/DisassemblyReport.h"
+#include "OrbitBase/ExecutablePath.h"
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/ReadFileToString.h"
+#include "capture_data.pb.h"
+
+using namespace std::string_view_literals;
+const std::string_view kMainFunctionInstructions =
+    "\x50\xbf\x04\x20\x40\x00\xe8\xe5\xfe\xff\xff\x31\xc0\x59\xc3\x90"sv;
+
+static void TestSimple(bool windows_line_endings) {
+  const std::filesystem::path file_path =
+      orbit_base::GetExecutableDir() / "testdata" / "line_info_test_binary";
+
+  auto program = orbit_object_utils::CreateElfFile(file_path);
+  ASSERT_TRUE(program.has_value()) << program.error().message();
+
+  constexpr uint64_t kAddressOfMainFunction = 0x401140;
+  ErrorMessageOr<orbit_grpc_protos::LineInfo> decl_line_info =
+      program.value()->GetDeclarationLocationOfFunction(kAddressOfMainFunction);
+  ASSERT_TRUE(decl_line_info.has_value()) << decl_line_info.error().message();
+
+  const std::filesystem::path source_file_path =
+      orbit_base::GetExecutableDir() / "testdata" / "LineInfoTestBinary.cpp";
+
+  ASSERT_TRUE(std::filesystem::exists(source_file_path));
+
+  ErrorMessageOr<std::string> source_file_contents_or_error =
+      orbit_base::ReadFileToString(source_file_path);
+  ASSERT_FALSE(source_file_contents_or_error.has_error());
+  std::string source_file_contents = std::move(source_file_contents_or_error.value());
+
+  if (windows_line_endings) {
+    absl::StrReplaceAll({{"\r\n", "\n"}}, &source_file_contents);
+    absl::StrReplaceAll({{"\n", "\r\n"}}, &source_file_contents);
```

Comment:
Maybe quick comment what this is about. It looks pretty useless at first glance. I assume the explanation is something like: "Make sure that all line endings are indeed windows line endings."

Reply:
Yes. Good point. I added a comment.